### PR TITLE
feat(instance): support scratch volumes in servers

### PR DESCRIPTION
**Pull request by Codelax**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request.
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Release note for [CHANGELOG](https://github.com/scaleway/scaleway-cli/blob/master/CHANGELOG.md):

```release-note

```

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -493,6 +493,8 @@ github.com/sabhiram/go-gitignore v0.0.0-20210923224102-525f6e181f06 h1:OkMGxebDj
 github.com/sabhiram/go-gitignore v0.0.0-20210923224102-525f6e181f06/go.mod h1:+ePHsJ1keEjQtpvf9HHw0f4ZeJ0TLRsxhunSI2hYJSs=
 github.com/scaleway/scaleway-sdk-go v1.0.0-beta.20.0.20230911082817-45ad1e03b1bb h1:symGjmphE602x27B1kY17r2BamxUaO5alk8y/Vw5N5M=
 github.com/scaleway/scaleway-sdk-go v1.0.0-beta.20.0.20230911082817-45ad1e03b1bb/go.mod h1:fCa7OJZ/9DRTnOKmxvT6pn+LPWUptQAmHF/SBJUGEcg=
+github.com/scaleway/scaleway-sdk-go v1.0.0-beta.21 h1:yWfiTPwYxB0l5fGMhl/G+liULugVIHD9AU77iNLrURQ=
```
Comment (remyleone) [inline go.sum:496]: go mod tidy

File: internal/namespaces/instance/v1/custom_server_create_test.go [hunk 2]
```diff
@@ -485,3 +486,29 @@ func Test_CreateServerErrors(t *testing.T) {
 		),
 	}))
 }
+
+func Test_CreateServerScratchStorage(t *testing.T) {
+	t.Run("Default scratch storage", core.Test(&core.TestConfig{
+		Commands: GetCommands(),
+		Cmd:      "scw instance server create type=H100-1-80G image=ubuntu_jammy_gpu_os_12 --debug",
```
Comment (remyleone) [inline internal/namespaces/instance/v1/custom_server_create_test.go:493]: is it normal to have the debug flag here?

Comment (Codelax) [reply internal/namespaces/instance/v1/custom_server_create_test.go]: It was for debugging purpose, requests are not logged unless this is added

File: internal/namespaces/instance/v1/custom_server_create.go [hunk 3]
```diff
@@ -473,7 +518,7 @@ func buildVolumes(api *instance.API, zone scw.Zone, serverName, rootVolume strin
 // Volumes definition must be through multiple arguments (eg: volumes.0="l:20GB" volumes.1="b:100GB")
 //
 // A valid volume format is either
-// - a "creation" format: ^((local|l|block|b):)?\d+GB?$ (size is handled by go-humanize, so other sizes are supported)
+// - a "creation" format: ^((local|l|block|b|s|scratch):)?\d+GB?$ (size is handled by go-humanize, so other sizes are supported)
```
Comment (quantumsheep) [inline internal/namespaces/instance/v1/custom_server_create.go:521]:
```suggestion
// - a "creation" format: ^((local|l|block|b|scratch|s):)?\d+GB?$ (size is handled by go-humanize, so other sizes are supported)
```

Follow the same order as the others

**Review by quantumsheep — CHANGES_REQUESTED** [review]

**Review by quantumsheep — APPROVED** [review]

**Pull request merged**